web/src/engine/keyboard/src/keyboards/loaders/domKeyboardLoader.ts

```diff
@@ -28,54 +29,44 @@ export class DOMKeyboardLoader extends KeyboardLoaderBase {
     this.performCacheBusting = cacheBust || false;
   }
 
-  protected loadKeyboardInternal(
-    uri: string,
-    errorBuilder: KeyboardLoadErrorBuilder,
-    id?: string
-  ): Promise<Keyboard> {
-    const promise = new ManagedPromise<Keyboard>();
-
-    if(this.performCacheBusting) {
+  protected async loadKeyboardBlob(uri: string, errorBuilder: KeyboardLoadErrorBuilder): Promise<Uint8Array> {
+    if (this.performCacheBusting) {
       uri = this.cacheBust(uri);
     }
 
+    let response: Response;
     try {
-      const document = this.harness._jsGlobal.document;
-      const script = document.createElement('script');
-      if(id) {
-        script.id = id;
-      }
-      document.head.appendChild(script);
-      script.onerror = (err: any) => {
-        promise.reject(errorBuilder.missingError(err));
-      }
-      script.onload = () => {
-        if(this.harness.loadedKeyboard) {
-          const keyboard = this.harness.loadedKeyboard;
-          this.harness.loadedKeyboard = null;
-          promise.resolve(keyboard);
-        } else {
-          promise.reject(errorBuilder.scriptError());
-        }
-      }
+      response = await fetch(uri);
+    } catch (e) {
+      throw errorBuilder.keyboardDownloadError(e);
+    }
 
-      // On the oldest mobile devices we support, Promise.finally may not actually exist.
-      // Fortunately... it's not that hard of an issue to work around.
-      // Note:  es6-shim doesn't polyfill Promise.finally!
-      promise.then(() => {
-        // It is safe to remove the script once it has been run (https://stackoverflow.com/a/37393041)
-        script.remove();
-      }).catch(() => {
-        script.remove();
-      });
+    if (!response.ok) {
+      throw errorBuilder.keyboardDownloadError(new Error(`HTTP ${response.status} ${response.statusText}`));
+    }
+
+    let buffer: ArrayBuffer;
+    try {
+      buffer = await response.arrayBuffer();
+    } catch (e) {
+      throw errorBuilder.invalidKeyboard(e);
+    }
```

Comment:
```suggestion
    const buffer: ArrayBuffer = await (async () => {
    try {
      return await response.arrayBuffer();
    } catch (e) {
      throw errorBuilder.invalidKeyboard(e);
    })();
```

Is this nicer? Hmm, it might be uglier.